app/controllers/application_controller.rb

```diff
@@ -43,4 +45,15 @@ def destroy_user_if_current_user
     @user = User.new
   end 
   
+  private
+  
+  def extract_locale_from_accept_language_header
+    language = request.env['HTTP_ACCEPT_LANGUAGE'].presence
+    if language
+      locale = HttpAcceptLangParser.parse(language)
+      logger.debug "* Locale set using accept_language to '#{locale}'"
+      locale.in?(User::LANGUAGES) ? locale : nil
```

Comment:
move this conditional logic to HttpAcceptLangParser, so all logic (knowledge) is centralized in a single place